File: server/src/main/kotlin/pt/isel/g20/unicommunity/board/model/Board.java
package pt.isel.g20.unicommunity.board.model;

import javax.persistence.*;

@Entity
public class Board {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;

    @Column(nullable = false)
    private String name;

    @Column
    private String description;

    public Board(){}

    public Board(String name, String description){
        this.name=name;
        this.description=description;
    }

    public Board(Long id, String name, String description){
        this.id = id;
        this.name = name;
        this.description = description;
    }

    public Board(String name){
        this.name=name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }
}

File: server/src/main/kotlin/pt/isel/g20/unicommunity/board/model/Board.kt
package pt.isel.g20.unicommunity.board.model

import pt.isel.g20.unicommunity.blackboard.model.Blackboard
import pt.isel.g20.unicommunity.forum.model.Forum
import javax.persistence.*

@Entity
class Board(
    @Column(nullable = false) var name: String,

    // TODO: Devia criar a relacao com o template e ter aqui uma referencia para o template?
    @Column var templateId: Long?,

    @Column var description: String? = null
) {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0

    @OneToOne
    @JoinColumn
    var forum: Forum? = null

    @OneToMany(fetch = FetchType.LAZY)
    @JoinColumn
    val blackBoards: MutableList<Blackboard> = mutableListOf()

    constructor() : this("", 0, null)
    constructor(name: String, templateId: Long) : this(name, templateId, null)
}
